torch/quantization/fx/fuse.py

```diff
@@ -0,0 +1,155 @@
+import torch
+from torch.quantization.fuse_modules import (
+    fuse_conv_bn,
+    fuse_conv_bn_relu,
+)
+
+from torch.fx import (
+    GraphModule,
+)
+
+from torch.fx.graph import (
+    Graph,
+    map_arg,
+)
+
+from torch.fx.symbolic_trace import (
+    DefaultDelegate,
+)
+
+from .pattern_utils import (
+    matches,
+    register_fusion_pattern,
+    get_fusion_patterns,
+)
+
+from .utils import _parent_name
+
+import copy
+
+# Fusion Patterns
+@register_fusion_pattern((torch.nn.BatchNorm2d, torch.nn.Conv2d))
```

Comment:
haven't looked at how this is used, but one question I have is if we can make the order of the pattern follow the order of the layers in the model (i.e. Conv-BN instead of BN-Conv)

Reply:
that is doable, but the matching is from later ops to former ops, it will confusing when people add new fusion patterns, I feel it would be better to keep the current order considering how matching works right now.

Reply:
cool, then maybe just a comment explaining this would work

Reply:
had a brief description in the quantize file, I'll move it to pattern_utils.py

Reply:
wanted to follow up on this, we should discuss defining patterns in the order that they occur in the graph, and changing the matching logic.  Reading through the current implementation, it's not intuitive that the order is reversed.

Reply:
sure, I'll change the order